cve_bin_tool/util.py

```diff
@@ -93,14 +93,19 @@ def __missing__(self, key: str) -> list[CVE] | set[str]:
         return self[key]
 
 
-def regex_find(lines: str, version_patterns: list[Pattern[str]]) -> str:
+def regex_find(
+    lines: str, version_patterns: list[Pattern[str]], ignore: list[Pattern[str]]
+) -> str:
     """Search a set of lines to find a match for the given regex"""
     new_guess = ""
 
     for pattern in version_patterns:
         match = pattern.search(lines)
         if match:
             new_guess = match.group(1).strip()
+            for i in ignore:
+                if str(i) in str(new_guess) or str(new_guess) in str(i):
+                    new_guess = ""
```

Comment:
I was wondering if we should remove the ignore patterns first and avoid the `new_guess=""` syntax, but upon thinking about it this seems computationally better since it means we only have to check them against a few lines.  Good thinking!